utils/report.c

```diff
@@ -726,31 +726,132 @@ static struct display_field field_##_func = {			\
 	.list    = LIST_HEAD_INIT(field_##_func.list)		\
 };
 
-#define FIELD_TIME(_id, _name, _field, _func, _header, _length)	\
+#define FIELD_TIME(_id, _name, _field, _func, _header)		\
 static void print_##_func(struct field_data *fd)		\
 {								\
 	struct uftrace_report_node *node = fd->arg;		\
 	print_time_unit(node->_field);				\
 }								\
-FIELD_STRUCT(_id, _name, _func, _header, _length)
+FIELD_STRUCT(_id, _name, _func, _header, 10)
 
-#define FIELD_CALL(_id, _name, _field, _func, _header, _length)	\
+#define FIELD_CALL(_id, _name, _field, _func, _header)		\
 static void print_##_func(struct field_data *fd)		\
 {								\
 	struct uftrace_report_node *node = fd->arg;		\
 	pr_out("%10"PRIu64 "", node->_field);			\
 }								\
-FIELD_STRUCT(_id, _name, _func, _header, _length)
-
-FIELD_TIME(REPORT_F_TOTAL_TIME, total, total.sum, total, "Total time", 10);
-FIELD_TIME(REPORT_F_TOTAL_TIME_AVG, total-avg, total.avg, total_avg, "Total avg", 10);
-FIELD_TIME(REPORT_F_TOTAL_TIME_MIN, total-min, total.min, total_min, "Total min", 10);
-FIELD_TIME(REPORT_F_TOTAL_TIME_MAX, total-max, total.max, total_max, "Total max", 10);
-FIELD_TIME(REPORT_F_SELF_TIME, self, self.sum, self, "Self time", 10);
-FIELD_TIME(REPORT_F_SELF_TIME_AVG, self-avg, self.avg, self_avg, "Self avg", 10);
-FIELD_TIME(REPORT_F_SELF_TIME_MIN, self-min, self.min, self_min, "Self min", 10);
-FIELD_TIME(REPORT_F_SELF_TIME_MAX, self-max, self.max, self_max, "Self max", 10);
-FIELD_CALL(REPORT_F_CALL, call, call, call, "Calls", 10);
+FIELD_STRUCT(_id, _name, _func, _header, 10)
```

Comment:
Can you please move this part into the previous commit?

Reply:
it's done. thank you.